Parse SVG CSS Style Sheets

adozenlines opened the pull request:
Integrated this project to provide support for recognizing embedded CSS style sheets.

Credit Original Stylesheet project
https://github.com/adozenlines/stylesheet

micahpearlman committed:
support image mode setting (except for stencil)

micahpearlman committed:
font color working

micahpearlman committed:
fixed crash when vbo isn't created

pholden committed:
adding xcode4 files to ignore to .gitignore

pholden committed:
add boost framework to my fork

micahpearlman committed:
rationalized the transform matrices

pholden committed:
removing prefix header (not used in this project)

pholden committed:
adding initial Test Android app ... needs a little cleanup but functional

pholden committed:
completed refactoring of android openvg implementation and test app (works with Android 2.2 tested on Motorola Droid X2

micahpearlman committed:
Merge branch 'master' of https://github.com/pholden/MonkVG

micahpearlman committed:
updated readme

micahpearlman committed:
dummy

micahpearlman committed:
Xcode 4.1 "modernization" of project

micahpearlman committed:
implemented tesselation iterations.

micahpearlman committed:
rough out of linear gradient paint.  just support for parameters.  not working.

micahpearlman committed:
working test method for generating gradient images.  still no working gradients.

micahpearlman committed:
support for different gradient spread modes.  still not working but test creates the linear gradient bitmap.

micahpearlman committed:
fixed memory leak in path generation

micahpearlman committed:
added a glPlatform.h file to localize the different platform GL includes

micahpearlman committed:
linear gradient implemented for fills!  Not working for strokes.

NEXT: image fill and radial gradients.

micahpearlman committed:
Edited README.md via GitHub

micahpearlman committed:
first working radial gradient.

bug: breaks images??

micahpearlman committed:
fixed issue where images couldn't be drawn at the same time as a gradient.

micahpearlman committed:
Merge branch 'master' of github.com:micahpearlman/MonkVG

micahpearlman committed:
updated readme

micahpearlman committed:
readme formatting fix

micahpearlman committed:
one more quick format fix for readme

Commit:
add support for windows platform - fix annoying warning under visual

micahpearlman committed:
small change to test ios renderer

micahpearlman committed:
fixed ios compile issue with missing APIENTRY define

micahpearlman committed:
added vincent richomme contribution in readme.md

micahpearlman committed:
Edited README.md via GitHub

micahpearlman committed:
Update README.md

micahpearlman committed:
Update README.md

talentless committed:
Added support for 2x3 matrix gradient paints

micahpearlman committed:
Merge pull request #3 from talentless/master

The base of gradient paints that match up with SWF data.

lukelutman committed:
Implemented support for VG_QUAD_TO

micahpearlman committed:
Merge pull request #6 from lukelutman/master

Support for VG_QUAD_TO

lukelutman committed:
Fixed XCode warnings.

micahpearlman committed:
Merge pull request #7 from lukelutman/master

Fixed XCode warnings

lukelutman committed:
Fixed invalid enum.

lukelutman committed:
Merge remote branch 'micahpearlman/master'

micahpearlman committed:
Merge pull request #8 from lukelutman/master

Fixed invalid enum.

lukelutman committed:
Updated iOS static library settings for better Xcode workspace integration.

Changed the 'Public Headers Folder Path' and changed the visibility of the headers so that
xcode puts the public headers where other projects in the workspace can find them (other
projects in the workspace no longer have to explicity add MonkVG to 'Header Search Paths').

See: http://www.blog.montgomerie.net/easy-xcode-static-library-subprojects-and-submodules

micahpearlman committed:
Merge pull request #9 from lukelutman/master

Updated iOS static library build settings

lukelutman committed:
Refactored include path from include/VG to include/MonkVG.

micahpearlman committed:
Merge pull request #10 from lukelutman/master

Refactored include path from include/VG to include/MonkVG

microcai committed:
add autotools build system, and fix builds on linux.

microcai committed:
add stub for vgFlush and vgFinish

microcai committed:
add stub for vgMask

microcai committed:
autotools checking more GL stuff like GLEW

micahpearlman committed:
Merge pull request #11 from microcai/master

autotools/libtools intergration patch

lukelutman committed:
Added support for dumping vertex data from a batch.

micahpearlman committed:
added gles2-bc

micahpearlman committed:
can compile shaders into the executable and read them.

micahpearlman committed:
converted over to using gles2-bc for GLES 1.1 ONLY!  Need to implement glColor stuff for the GLES 2.0 version

micahpearlman committed:
separated the MonkVG specific example code into separate class/file

micahpearlman committed:
added the ability to retrieve the backing width and height (aka screen size).

micahpearlman committed:
first run with OpenGL es 2.0.  Need to fix glColor* stuff.

micahpearlman committed:
initial support for simulating glColor4f in GLES 2.0

micahpearlman committed:
fixed bug with glColor in GLES 2.0 where the color was not modulated on a texture

micahpearlman committed:
updated README

micahpearlman committed:
updated boost library framework to support armv7s

micahpearlman committed:
fixer weird problem with being able to link MonkVG

micahpearlman committed:
Now checking if the current shader is what we expect it to be.

micahpearlman committed:
Merge commit '6d5d1e03aca38bc45c0329e1d3a73d9733873eb9'

micahpearlman committed:
replaced <cstring> with the C <stddef.h>

Commit:
Fixed commit "replaced <cstring> with the C <stddef.h>".
The commit c94e6cfa86d621b503583369031c76798f270c04 removed the <cstring> include but didn't replace it with <stddef.h> and as a result the compiler throws an error for the iOS project as it can't find the definition of size_t.

micahpearlman committed:
Merge pull request #14 from colinmccune/master

Fixed "replaced <cstring> with the C <stddef.h>" commit.

micahpearlman committed:
OSX Xcode project now compiles the MonkVG library.

**WARNING**

Have not actually tested the library to see if it actually works!

Commit:
Bootstrap script added for to create autotools files ready for ./configure && make
Required header included.
Required include paths added.

Commit:
Build fixes for Linux/GLES2.
Integration of gles2-bc and glu into autotools build system.
Fixed compliance issues with shaders.
Updated shader include files and added script to automate it.
Added USE_OPENGL macro for switching between desktop and embedded GL on Linux.
Added USE_SYSTEM_FMEMOPEN macro for using system fmemopen.
Added PLATFORM_IS_MESA macro to avoid using extensions that are not supported
by MesaGL on Linux.

Commit:
Backslash missing at end of line.

Commit:
Moved autotools to their own path.
Added Android build.

Commit:
Android build.

Removed fmemopen in favour of slightly higher-level shader-from-memory
integration.
It was always going to be fundamentally broken due to its reliance on
fpos_t, an opaque datatype. New method sacrifices nothing and is overall
less code.
Fixes to shaders; in particular for broken shader-preprocessor
implementations on Android (#elif doesn't work properly).

Commit:
Removed unnecessary boost files from test project, and made it defer to proper Android project.

micahpearlman committed:
Merge pull request #15 from gavofyork/master

Android & Linux builds and a few cleanups & fixes.

micahpearlman committed:
fixed iOS project.

Added Gav Wood as a contributor.

codedread committed:
Fix MonkVG-OSX project by removing references to fmemopen

micahpearlman committed:
Merge pull request #16 from codedread/master

Removed fmemopen references from MonkVG-OSX project

gaffo committed:
Drop autoconf requirment to 2.88 which is default on ubuntu 12.04

micahpearlman committed:
Merge pull request #17 from gaffo/drop_ac_to288

Drop autoconf requirment to 2.88 which is default on ubuntu 12.04

micahpearlman committed:
updated iOS projects to latest xcode

micahpearlman committed:
Merge branch 'master' of github.com:micahpearlman/MonkVG

micahpearlman committed:
updated boost library to Xcode 5

micahpearlman committed:
updated boost with regex libraries

rhcad committed:
Remove boost.framework and replace boost::ptr_map with hash_map.

rhcad committed:
Avoid compile warnnings.

micahpearlman committed:
got rid of the boost framework

Commit:
Avoid memory leak.

rhcad committed:
Merge pull request #1 from micahpearlman/master

got rid of the boost framework

rhcad committed:
Output library both for iPad and iPhone.

rhcad committed:
Implement VG_QUAD_TO in buildStroke(). Enable to free GL context.

Reorder Initialize() statement in vgCreateContextMNK() to avoid a warning about zero width.
Enable to free GL context in vgDestroyContextMNK(), so GL views can be created multiple times.

rhcad committed:
Remove unnecessary boost header files for Android.

rhcad committed:
Fix compiler errors for Android: support for C++11, 'std::malloc' was not declared.

rhcad committed:
Fix the Android test application.
1. Don't link libOpenVG for the libOpenVGU static library, or fail to compile libmonkvg.so.
2. Call vgCreateContextMNK and vgResizeSurfaceMNK to create gl() object.
3. Call glView.setEGLContextClientVersion(2) to use OpenGL 2.0.

But I can't see any shape except the green screen on Moto pad, invalid transform?

rhcad committed:
Fix buildFatLineSegment() to avoid too bold width.

rhcad committed:
Fix the warning of static library need't ld.

micahpearlman committed:
Merge pull request #24 from rhcad/master

Implement VG_QUAD_TO in buildStroke(). Enable to free GL context and create GL view multiple times..

micahpearlman committed:
fixed bug with main shader multiplying color.

doppioslash committed:
Added Android build howto

doppioslash committed:
Switched to c++_static

micahpearlman committed:
Merge pull request #30 from doppioslash/android

Added Android build howto

jacereda committed:
Fix disjoint contours

micahpearlman committed:
Merge pull request #31 from jacereda/disjoint-contours

Fix disjoint contours

jacereda committed:
Allow VG_INVALID_HANDLE argument to vgSetPaint().
Set error on bad paintMode to vgSetPaint().

micahpearlman committed:
Merge pull request #32 from jacereda/master

Fix vgSetPaint()

jacereda committed:
Fix non-standard behaviour for gradient stops

jacereda committed:
Bad literal

jacereda committed:
Implement vgGetPaint()

jacereda committed:
Implement vgPathTransformedBounds()

micahpearlman committed:
Merge pull request #33 from jacereda/fix-stops

Fix non-standard behaviour for gradient stops

micahpearlman committed:
Merge pull request #34 from jacereda/vggetpaint

Implement vgGetPaint()

micahpearlman committed:
Merge pull request #35 from jacereda/vgpathtransformedbounds

Implement vgPathTransformedBounds()

jacereda committed:
- Final bounding box is computed at the end of buildFillIfDirty() instead of draw().
- Made some methods private.
- v3_t empty constructor no longer initializes elements.
- Calls to gluTessVertex() are now less verbose.

jacereda committed:
- Shouldn't be necessary to keep track of closeTo.
- numCoords unused.

jacereda committed:
calcStops() should now be closer to the standard.

micahpearlman committed:
Merge pull request #37 from jacereda/fix-stops

calcStops() should now be closer to the standard.

micahpearlman committed:
Merge pull request #36 from jacereda/glpath-enhancements

Glpath enhancements

jacereda committed:
Less verbose error handling

micahpearlman committed:
Merge pull request #38 from jacereda/less-verbose-errors

Less verbose error handling

Commit:
parse embedded style sheets

adozenlines committed:
Update .gitmodules

Commit:
Null termination

Commit:
Removed sample svg content

Commit:
Parse SVG CSS style

micahpearlman commented:
Thanks!  I'll take a look at this closer sometime this week.  If you want you can update the readme.md with a credit to your contribution.

adozenlines commented:
Will do.

Thanks.

> On Jul 21, 2015, at 1:33 PM, Micah Pearlman <notifications@github.com> wrote:
> 
> Thanks! I'll take a look at this closer sometime this week. If you want you can update the readme.md with a credit to your contribution.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/micahpearlman/MonkSVG/pull/31#issuecomment-123410756>.
>

Commit:
Readme.md Updated
- Credits & Copyright

adozenlines commented:
You don’t happen to have a modern iOS implementation of MonkSVG & GLKit?

Sean.

> On Jul 21, 2015, at 1:38 PM, Sean Batson <sean.batson@gmail.com> wrote:
> 
> Will do.
> 
> Thanks.
> 
>> On Jul 21, 2015, at 1:33 PM, Micah Pearlman <notifications@github.com <mailto:notifications@github.com>> wrote:
>> 
>> Thanks! I'll take a look at this closer sometime this week. If you want you can update the readme.md with a credit to your contribution.
>> 
>> —
>> Reply to this email directly or view it on GitHub <https://github.com/micahpearlman/MonkSVG/pull/31#issuecomment-123410756>.
>> 
>

micahpearlman commented:
Not with GLKit unfortunately. Are there issues?



> On Jul 21, 2015, at 6:13 PM, Sean Batson <notifications@github.com> wrote:
> 
> You don’t happen to have a modern iOS implementation of MonkSVG & GLKit?
> 
> Sean.
> 
> > On Jul 21, 2015, at 1:38 PM, Sean Batson <sean.batson@gmail.com> wrote:
> > 
> > Will do.
> > 
> > Thanks.
> > 
> >> On Jul 21, 2015, at 1:33 PM, Micah Pearlman <notifications@github.com <mailto:notifications@github.com>> wrote:
> >> 
> >> Thanks! I'll take a look at this closer sometime this week. If you want you can update the readme.md with a credit to your contribution.
> >> 
> >> —
> >> Reply to this email directly or view it on GitHub <https://github.com/micahpearlman/MonkSVG/pull/31#issuecomment-123410756>.
> >> 
> > 
> 
> —
> Reply to this email directly or view it on GitHub.
>

adozenlines commented:
Shapes with alpha being drawing without the alpha …


> On Jul 22, 2015, at 1:01 AM, Micah Pearlman <notifications@github.com> wrote:
> 
> Not with GLKit unfortunately. Are there issues?
> 
> 
> 
> > On Jul 21, 2015, at 6:13 PM, Sean Batson <notifications@github.com> wrote:
> > 
> > You don’t happen to have a modern iOS implementation of MonkSVG & GLKit?
> > 
> > Sean.
> > 
> > > On Jul 21, 2015, at 1:38 PM, Sean Batson <sean.batson@gmail.com> wrote:
> > > 
> > > Will do.
> > > 
> > > Thanks.
> > > 
> > >> On Jul 21, 2015, at 1:33 PM, Micah Pearlman <notifications@github.com <mailto:notifications@github.com>> wrote:
> > >> 
> > >> Thanks! I'll take a look at this closer sometime this week. If you want you can update the readme.md with a credit to your contribution.
> > >> 
> > >> —
> > >> Reply to this email directly or view it on GitHub <https://github.com/micahpearlman/MonkSVG/pull/31#issuecomment-123410756>.
> > >> 
> > > 
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> > 
> —
> Reply to this email directly or view it on GitHub <https://github.com/micahpearlman/MonkSVG/pull/31#issuecomment-123562776>.
>

Commit:
Code Cleanup
 - Stylesheet reference

Commit:
Object drawing sequence of operations fixes the application of the alpha when using embedded style sheets
 - Fill or Stroke then apply the attributes of those two verbs

Commit:
Search for boost recursively

Commit:
Merge branch 'master' of https://github.com/adozenlines/MonkSVG

* 'master' of https://github.com/adozenlines/MonkSVG:
  Readme.md Updated - Credits & Copyright
  Parse SVG CSS style

Commit:
Parse viewBox attribute

Commit:
Parse embedded SVG CSS style sheet

adozenlines committed:
Update README.md

Commit:
Monk Changes
  Additional SVG stroke and line options

Commit:
Added shapes line & polyline
 to parsing flow

Commit:
Modern settings applied (Xcode Thing)

Commit:
Modern settings applied (Xcode Thing)

Commit:
Rotation XYZ

Commit:
Attempt to produce smooth lines
 might have to look at map box gl

Commit:
Xcode version stamp change

Commit:
Xcode version stamp

Commit:
Sub project updates

Commit:
Subprojects

adozenlines committed:
Update README.md

Commit:
Initial changes for Line styling at the path level

Commit:
Initial changes for line styling at the path level

Commit:
Public API calls for line path styling (Join, Cap, Miter)

Commit:
seem like a waste of time ...

Commit:
Changes to the gl patterns
todo proper line shading ...

Commit:
Merge branch 'master' of https://github.com/adozenlines/MonkVG

# By Sean Batson
# Via Sean Batson
* 'master' of https://github.com/adozenlines/MonkVG:
  Update README.md

Commit:
got rid of boost dependency
added temporary fix for bug: path fill paint was created when setting fill opacity even if fill:none was set

Commit:
Removed "useless" files and third parties.

Commit:
Modded files to use the latest 3rd parties. Also removed boost usage.

Commit:
Added submodule third parties

Commit:
Now with project that uses all third parties

Commit:
Merge branch 'master' of https://github.com/VasilievMB/MonkSVG

Commit:
Sakamura/MonkVG

Commit:
Removed MonkVG as there's no need to get a VG implementation in particular

Commit:
Added OpenVG, StyleSheet and tinyxml2 submodules

Commit:
Deleted files migrated to thirdparty and examples, and other "useless" files.

Commit:
Added MonkSVG Xcode project

Commit:
mkSVG compiles with the new libraries

Commit:
Added missing file. Fixed a few nonworking elements in order to play the test adequately.

Commit:
Fixing noncompiling elements. Fixed tesselation vertexes (preliminary - this is non optimal)

sakamura committed:
All changes Sakamura did

sakamura committed:
Reflects Sakamura version

sakamura committed:
Update README.md

sakamura committed:
Update README.md

sakamura committed:
Update README.md

Commit:
Now uses sqlite to reduce polygons of batches (first pass)

Commit:
Heavy optimization - first crude pass removing all gl files and migrating them to mk.

Commit:
Reduced virtuels

Commit:
Reduced virtuels

Commit:
Keep on cleaning up MonkVG. Renamed I* prefixes to MK* prefixes as they aren't interfaces anymore.

Commit:
Added cpp_btree submodule

Commit:
MonkVG: Work In Progress. Added back VG_STROKE_PATH. Removed GLE backward compatibility library. Removed SQLite, uses cpp_btree instead. Multiple optimizations. Now renders the batch without optimization but correctly.

Commit:
Merged MonkVG as a subtree. Will serve as base to migrate code to MonkSVG and remove OpenVG dependencies.

Commit:
MonkSVG->SakaSVG

sakamura committed:
Updated BSD-3 clause & Copyright holder

Added myself as copyright. Used latest wording version of the same licensing (not a different license than before). Very obviously kept previous copyright notice from MonkVG & MonkSVG.

sakamura committed:
SakaSVG Readme

sakamura committed:
More contributors from MonkVG

sakamura committed:
Adds licensing mapbox-gl-native files

sakamura committed:
Added mapbox copyright

sakamura committed:
Added mapbox-gl copyright

sakamura committed:
Added mapbox-gl-native as contributors

Commit:
Removed MonkVG ThirdParty. Migrated to SakaSVG ThirdParty.

Commit:
Removed MonkVG's Readme and License (now part of SakaSVG)

sakamura committed:
Added MonkVG Todo's

Commit:
Merge remote-tracking branch 'origin/master'

sakamura committed:
Added Other Projects section

Commit:
Moved mkOpenVG_SVG into mkSVG

Commit:
Updated projects to latest Xcode recommended settings

Commit:
Cleanup : drawing is only made through mkBatch. No more ES1.

Commit:
Migrated mkCommon to mkSVGHandler. Added SakaSVG base class. Removed all vg* function calls (now made directly to the classes). Assumes drawing is done through batch exclusively. No more BaseClass. No more virtuals.

Commit:
Now uses GLM instead of files borrowed from Mapbox.

Commit:
Updated to latest third parties. Added GLM.

sakamura committed:
Update README.md

sakamura committed:
Update README.md

sakamura committed:
Not using Mapbox software anymore.

Commit:
Removed VGU

Commit:
Removed OpenVG dependency

Commit:
Removed OpenVG

sakamura committed:
Update README.md

Commit:
More migration to glm. Tempanently remove line styles until completed.

Commit:
We want to draw the batch only, not anything else.

Commit:
SakaSVG isa mkBatch

Commit:
Updated licensing src files. Now only exports SakaSVG. mkBatch is in mkSVG.

Commit:
Added a simple GL Manager to reuse vao, vbo & ebo.

Commit:
GL Manager is in its own file. Added programs & shaders. Added position matrix (for now it just moves it full screen). Hard-coded basic shader to code.

Commit:
Uses TESS_CONSTRAINED_DELAUNAY_TRIANGLES.

Commit:
More cleanup, as mk doesn't draw anymore, so all gl stuff is useless there.

Commit:
Move everything mk-related to a mk folder.

Commit:
Added polyclip submodule

Commit:
Migrated Polyclip to h/inl paradigm for templatization.

Commit:
Renamed bucket alloc additional functions to be tess*.

Commit:
Uses latest glm. Includes polyclip (but unused yet)

sakamura committed:
Added polyclip to third parties.

Commit:
Converted libTess from h to cpp, with a Tess namespace.

Commit:
Removing Tess projection step. Now uses s,t directly.

Commit:
Merge branch 'master' of https://github.com/sakamura/SakaSVG

Commit:
Added back polyclip (xcode git error)

Commit:
Latest libtess2 changes

Commit:
Remove Tess ability to longjmp or returning 0/1 when allocations fail.

Commit:
C++ification of Tess: now has c++ class overrides, no TESSalloc and c++ version of bucketalloc. FWIW, bucketalloc will probably be out eventually as C++ prefers to encapsulate complexity on the container, not the struct itself. Baby steps.

Commit:
C++ification of Tess: tess.h,.cpp and sweep.cpp are in Tesselator class. Cleaned up defines, NULL, TRUE/FALSE, geom.

Commit:
libtess2 c++ification

Commit:
Updating projects to latest Xcode

Commit:
Now compiles

Commit:
Normalized projects for latest iOS/XCode with heavy statical analysis

Commit:
Unleak, faster

Commit:
Now with much increased speed (first part)

Commit:
Warning Cleanup

Commit:
More logging. Fixed VG_MOVE that didn't move relatively

Commit:
First draft of Cubic to Quad. Will need the GPU implementation and shader soon.

Commit:
Fixed contour beginning and ending to be consistent with standard

sakamura committed:
Upgrade to latest Xcode

sakamura committed:
Updating projects

sakamura committed:
Fixing using libtess3.

sakamura committed:
Faster string parsing using boost. Fox is in the hen, eventually, we should rewrite the parser to use qi exclusively.

sakamura committed:
Standardized projects to latest iOS version for now.

sakamura committed:
WIP improvements
- No more need for Polyclip
- Now uses CubicToQuad library
- Vertices are now stored in GLM format, and stored dynamically in one single step through LibTess3
- Have a previous Vertex keeper while parsing, so we can modify previous one and add up requested quadratic fun
- For now, keeping a dynamic library of all points. This is what is taking the longest, even if b-tree is really efficient. So we'll have to add up options later on.
- Don't use the library headers, it's a nightmare to develop with two copies of every headers.
- TODO: For now, I have a thirdparty/ include path but there must be a better way.
- Read SVG from a XML document instead of a file path, so we can keep the xml doc in memory.

sakamura committed:
Update README.md

sakamura committed:
Updated subproject commits for SakaSVG

sakamura committed:
More subproject updates

sakamura committed:
Latest GLM requires experimental features enabling

sakamura committed:
Latest tinyxml2, latest glm

sakamura committed:
Fixed issues with latest GLM

sakamura committed:
Upgraded Xcode projects

adozenlines committed:
Merge pull request #1 from sakamura/master

Projects updates
